addons/im_livechat/static/src/core/web/livechat_conversation_tag_edit.js

```diff
@@ -0,0 +1,121 @@
+import { Component, onWillStart, useEffect, useState } from "@odoo/owl";
+
+import { useAutofocus, useService } from "@web/core/utils/hooks";
+import { useSequential } from "@mail/utils/common/hooks";
+import { useDebounced } from "@web/core/utils/timing";
+import { escapeRegExp } from "@web/core/utils/strings";
+
+export class ConversationTagEdit extends Component {
+    static components = {};
+    static props = ["thread", "autofocus?", "close?", "selectedTags?"];
+    static template = "im_livechat.ConversationTagEdit";
+
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        this.store = useService("mail.store");
+        this.inputRef = useAutofocus();
+        this.sequential = useSequential();
+        this.state = useState({
+            selectableTags: [],
+            selectedTags: [...(this.props.selectedTags || [])],
+            searchStr: "",
+        });
+        this.debouncedFetchConversationTags = useDebounced(
+            this.fetchConversationTags.bind(this),
+            250
+        );
+        onWillStart(() => {
+            this.fetchConversationTags();
+        });
+        useEffect(
+            () => {
+                this.debouncedFetchConversationTags();
+            },
+            () => [this.state.searchStr]
+        );
+        useEffect(
+            () => {
+                this.sequential(async () => {
+                    const selectedTagIds = this.state.selectedTags.map((tag) => tag.id).sort();
+                    await this.orm.write("discuss.channel", [this.props.thread.id], {
+                        livechat_conversation_tag_ids: selectedTagIds.length
```

Comment:
Do we actually need to send the full list?
As posted in another comment, it might be better to explicitly send what should be added and what should be removed.